include/Basis/LibcintInterfacer.hpp

```diff
@@ -0,0 +1,232 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#ifndef LibcintInterfacer_hpp
+#define LibcintInterfacer_hpp
+
+#include "Operator/OneElectronOperator.hpp"
+#include "Operator/TwoElectronOperator.hpp"
+#include "Molecule.hpp"
+#include "Basis/ShellSet.hpp"
+
+#include <functional>
+
+
+
+extern "C" {
+
+#include <cint.h>
+
+
+/*
+ *  FUNCTIONS THAT AREN'T INSIDE <cint.h>
+ */
+int cint1e_ovlp_cart(double* buf, int* shls, int* atm, int natm, int* bas, int nbas, double* env);  // overlap
+int cint1e_kin_cart(double* buf, int* shls, int* atm, int natm, int* bas, int nbas, double* env);  // kinetic energy
+int cint1e_nuc_cart(double* buf, int* shls, int* atm, int natm, int* bas, int nbas, double* env);  // nuclear attraction energy
+int cint1e_r_cart(double* buf, int* shls, int* atm, int natm, int* bas, int nbas, double* env);  // dipole integrals, origin at zero
+
+
+}  // extern "C"
+
+
+
+
+namespace GQCP {
+
+
+/**
+ *  Forward declaration
+ */
+class LibcintInterfacer;
+
+
+
+/*
+ *  WRAPPERS AROUND C-STYLE LIBCINT ARRAYS
+ */
+namespace libcint {
+
+
+static constexpr int atm_slots = ATM_SLOTS;
+static constexpr int bas_slots = BAS_SLOTS;
```

Comment:
Idem, + Also why is `bas_slots ` declared while `BAS_SLOTS ` is still used in the gqcp code.